kddeployqt.py

```diff
@@ -0,0 +1,274 @@
+    if 'QT_INSTALL_LIBS' in line:
+        libs_dir = line.split(':')[1]
+    if 'QT_INSTALL_QML' in line:
+        qml_dir = line.split(':')[1]
+    if 'QT_HOST_BINS' in line:
+        host_bins = line.split(':')[1]
+    if 'QT_HOST_LIBEXECS' in line:
+        host_libexecs = line.split(':')[1]
+
+executables = [args.executable]
+
+if args.wayland:
+    args.extra_plugins.append('wayland-decoration-client')
+    args.extra_plugins.append('wayland-graphics-integration-client')
+    args.extra_plugins.append('wayland-shell-integration')
+
+def find_relative_path(path):
+    for root in args.qml_root_path:
+        if path.startswith(root):
+            p = path.replace(root, '')
+            if (p.startswith('/')):
+                p = p[1:]
+            return p
+    raise Exception('Could not find relative path for ' + path)
+
+qml_plugins = []
+# run qmlimportscanner to get the qml dependencies if qml_root_path exists
+if args.qml_root_path:
+    scanner = [os.path.join(host_libexecs, 'qmlimportscanner'), '-importPath', qml_dir]
+    for path in args.qml_root_path:
+        scanner.append('-rootPath')
+        scanner.append(path)
+
+    for path in args.qml_import_path:
+        scanner.append('-importPath')
+        scanner.append(path)
+    imports = json.loads(subprocess.check_output(scanner).decode('utf-8'))
+    for imp in imports:
+        if not 'path' in imp:
+            print('Skipping ' + imp['name'] + ' as it is not found')
+            continue
+        path = imp['path']
+        if not os.path.exists(path):
+            print('Skipping ' + imp['name'] + ' as it is not found')
+            continue
+
+        try:
+            out_path = os.path.join(out_qml_dir, imp['relativePath'] if 'relativePath' in imp else find_relative_path(path))
+        except Exception as e:
+            print(e)
+            continue
+        if (args.verbose):
+            print('Copying from ' + path + ' to ' + out_path)
+        if os.path.isdir(path):
+            shutil.copytree(path, out_path, dirs_exist_ok=True)
+        else:
+            os.makedirs(os.path.dirname(out_path), exist_ok=True)
+            shutil.copyfile(path, out_path)
+        if 'plugin' in imp:
+            if (args.verbose):
+                print('Found plugin ' + imp['plugin'])
+            qml_plugins.append('lib' + imp['plugin'] + '.so')
+
+    # scan for all .so.* files in qml folder
+    for root, dirs, files in os.walk(out_qml_dir):
+        for file in files:
+            if (file in qml_plugins):
+                if (args.verbose):
+                    print('Add ' + file + ' to the executables')
+                executables.append(os.path.join(root, file))
+
+
+def copy_plugins(plugins):
+        for plugin in plugins:
+            from_path = os.path.join(plugins_dir, plugin)
+            if (os.path.exists(from_path)):
+                if (args.verbose):
+                    print('\t\tcopying ' + from_path + ' to ' + out_plugins_dir)
```

Comment:
I think format strings with placeholders would be more readable here and elsewhere.